pkg/provider/aci_test.go

```diff
@@ -1785,5 +1785,78 @@ func TestGetContainerLogs(t *testing.T) {
 
 		})
 	}
+}
+
+func TestGetGPUSKU(t *testing.T) {
+	podName := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	aciMocks := createNewACIMock()
+
+	provider, err := createTestProvider(aciMocks, NewMockConfigMapLister(mockCtrl),
+		NewMockSecretLister(mockCtrl), NewMockPodLister(mockCtrl))
+	if err != nil {
+		t.Fatal("failed to create the test provider", err)
+	}
 
+	cases := []struct {
+		description   string
+		gpuSkus       []azaciv2.GpuSKU
+		desiredSku    string
+		expectedError error
+	}{
+		{
+			description:   "gpuTypeAnnotation is not set but ACI provides gpusku",
+			gpuSkus:       []azaciv2.GpuSKU{azaciv2.GpuSKUK80, azaciv2.GpuSKUP100},
+			desiredSku:    "",
+			expectedError: nil,
+		},
+		{
+			description:   "gpuTypeAnnotation is set and the desired sku is supported by ACI",
+			gpuSkus:       []azaciv2.GpuSKU{azaciv2.GpuSKUK80, azaciv2.GpuSKUP100},
+			desiredSku:    "P100",
```

Comment:
Is it possible to use the pre-defined consts i.e., `azaciv2.GpuSKUP100` in all desiredSKU??